components/SegmentERC721TokenGating.tsx

```diff
@@ -3,99 +3,136 @@
 
 import type React from "react";
 import type { SegmentCookieMetaProps } from "@/components/types/CookieTypes";
-import { useToken } from "wagmi";
+import { useReadContracts } from "wagmi";
 
 import {
-	FormControl,
-	FormDescription,
-	FormField,
-	FormItem,
-	FormLabel,
-	FormMessage,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
 } from "@/components/ui/form";
 import { Input } from "@/components/ui/input";
 
 import {
-	Collapsible,
-	CollapsibleContent,
-	CollapsibleTrigger,
+  Collapsible,
+  CollapsibleContent,
+  CollapsibleTrigger,
 } from "@/components/ui/collapsible";
+import { TokenInput } from "./ui/token-input";
+import { erc20Abi } from "viem";
 
 const SegmentERC721TokenGating: React.FC<SegmentCookieMetaProps<any>> = ({
-	form,
+  form,
 }) => {
-	const {
-		control,
-		formState: { isValid },
-		watch,
-	} = form;
+  const {
+    control,
+    formState: { isValid },
+    watch,
+    setValue,
+    clearErrors,
+    setError,
+  } = form;
 
-	const { data: erc721Token } = useToken({
-		address: watch("erc721Token") as `0x${string}`,
-	});
+  const { data: token } = useReadContracts({
+    allowFailure: false,
+    contracts: [
+      {
+        address: watch("erc721Token") as `0x${string}`,
```

Comment:
Avoid typecasting if possible. Import Address from wagmi/viem to get the type.
```suggestion
        address: watch("erc721Token") as Address,
```